ui/components/app/edit-gas-fee-popover/edit-gas-item/useGasItemFeeDetails.js

```diff
@@ -0,0 +1,126 @@
+
+export const useGasItemFeeDetails = (priorityLevel) => {
+  const {
+    editGasMode,
+    estimateUsed,
+    gasFeeEstimates,
+    gasLimit,
+    maxFeePerGas: maxFeePerGasValue,
+    maxPriorityFeePerGas: maxPriorityFeePerGasValue,
+    transaction,
+  } = useGasFeeContext();
+  const [estimateGreaterThanGasUse, setEstimateGreaterThanGasUse] = useState(
+    false,
+  );
+  const advancedGasFeeValues = useSelector(getAdvancedGasFeeValues);
+
+  let maxFeePerGas;
+  let maxPriorityFeePerGas;
+  let minWaitTime;
+
+  const { dappSuggestedGasFees } = transaction;
+
+  if (gasFeeEstimates?.[priorityLevel]) {
+    maxFeePerGas = gasFeeEstimates[priorityLevel].suggestedMaxFeePerGas;
+    maxPriorityFeePerGas =
+      gasFeeEstimates[priorityLevel].suggestedMaxPriorityFeePerGas;
+  } else if (
+    priorityLevel === PRIORITY_LEVELS.DAPP_SUGGESTED &&
+    dappSuggestedGasFees
+  ) {
+    maxFeePerGas = hexWEIToDecGWEI(
+      dappSuggestedGasFees.maxFeePerGas || dappSuggestedGasFees.gasPrice,
+    );
+    maxPriorityFeePerGas = hexWEIToDecGWEI(
+      dappSuggestedGasFees.maxPriorityFeePerGas || maxFeePerGas,
+    );
+  } else if (priorityLevel === PRIORITY_LEVELS.CUSTOM) {
+    if (estimateUsed === PRIORITY_LEVELS.CUSTOM) {
+      maxFeePerGas = maxFeePerGasValue;
+      maxPriorityFeePerGas = maxPriorityFeePerGasValue;
+    } else if (advancedGasFeeValues) {
+      maxFeePerGas =
+        gasFeeEstimates.estimatedBaseFee *
+        parseFloat(advancedGasFeeValues.maxBaseFee);
+      maxPriorityFeePerGas = advancedGasFeeValues.priorityFee;
+    }
+  } else if (
+    priorityLevel === PRIORITY_LEVELS.MINIMUM &&
+    transaction.previousGas
+  ) {
+    maxFeePerGas = hexWEIToDecGWEI(
+      addTenPercent(transaction.previousGas?.maxFeePerGas),
+    );
+    maxPriorityFeePerGas = hexWEIToDecGWEI(
+      addTenPercent(transaction.previousGas?.maxPriorityFeePerGas),
+    );
+  }
+
+  const { waitTimeEstimate } = useCustomTimeEstimate({
+    gasFeeEstimates,
+    maxFeePerGas,
+    maxPriorityFeePerGas,
+  });
+
+  if (gasFeeEstimates[priorityLevel]) {
+    minWaitTime =
+      priorityLevel === PRIORITY_LEVELS.HIGH
+        ? gasFeeEstimates?.high.minWaitTimeEstimate
+        : gasFeeEstimates?.low.maxWaitTimeEstimate;
+  } else {
+    minWaitTime = waitTimeEstimate;
+  }
+
+  const hexMaximumTransactionFee = maxFeePerGas
+    ? getMaximumGasTotalInHexWei({
+        gasLimit: decimalToHex(gasLimit),
+        maxFeePerGas: decGWEIToHexWEI(maxFeePerGas),
+      })
+    : null;
+
+  useEffect(() => {
+    // For cancel and speed-up medium / high option is disabled if
+    // gas used in transaction + 10% is greater tham estimate
+    if (
+      (editGasMode === EDIT_GAS_MODES.CANCEL ||
+        editGasMode === EDIT_GAS_MODES.SPEED_UP) &&
+      (priorityLevel === PRIORITY_LEVELS.MEDIUM ||
+        priorityLevel === PRIORITY_LEVELS.HIGH)
+    ) {
+      const estimateGreater = !gasEstimateGreaterThanGasUsedPlusTenPercent(
```

Comment:
Does this need to be kept in state and updated via a `useEffect`? What's the harm in re-calculating this every time this hook is run?

Reply:
It should be re-calculated as effect of `gasFeeEstimates` changing.